src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ConfirmationService.kt

```diff
@@ -70,85 +65,101 @@ class ConfirmationService(
     return response
   }
 
-  private fun admitOffender(
-    confirmArrivalDetail: ConfirmArrivalDetail,
-    moveId: String,
-    prisonNumber: String
-  ): ConfirmArrivalResponse = when (confirmArrivalDetail.movementReasonCode) {
-    in RECALL_MOVEMENT_REASON_CODES -> recallOffender(confirmArrivalDetail, moveId, prisonNumber)
-    else -> admitOffenderOnNewBooking(confirmArrivalDetail, moveId, prisonNumber)
+  fun confirmArrival(confirmation: Confirmation): ConfirmArrivalResponse {
+    if (confirmation.isNewToPrison) {
+      return createAndAdmitOffender(confirmation)
+    }
+
+    val prisoner = prisonerSearchService.getPrisoner(confirmation.prisonNumber!!)
+
+    with(prisoner) {
+      return if (isCurrentPrisoner) throw IllegalArgumentException("Confirming arrival of active prisoner: '$prisonNumber' is not supported.")
+      else admitOffender(confirmation, prisonNumber)
+    }
   }
 
-  private fun admitOffenderOnNewBooking(
-    confirmArrivalDetail: ConfirmArrivalDetail,
-    moveId: String,
-    prisonNumber: String
-  ): ConfirmArrivalResponse {
-    val inmateDetail = prisonService.admitOffenderOnNewBooking(prisonNumber, confirmArrivalDetail)
+  private fun admitOffender(confirmation: Confirmation, prisonNumber: String): ConfirmArrivalResponse =
+    when (confirmation.movementReasonCode) {
+      in RECALL_MOVEMENT_REASON_CODES -> recallOffender(confirmation, prisonNumber)
+      else -> admitOffenderOnNewBooking(confirmation, prisonNumber)
+    }
 
-    confirmedArrivalRepository.save(
-      ConfirmedArrival(
-        movementId = moveId,
-        prisonNumber = prisonNumber,
-        timestamp = LocalDateTime.now(clock),
-        arrivalType = ArrivalType.NEW_BOOKING_EXISTING_OFFENDER,
-        prisonId = confirmArrivalDetail.prisonId!!,
-        bookingId = inmateDetail.bookingId,
-        arrivalDate = confirmArrivalDetail.bookingInTime?.toLocalDate() ?: LocalDate.now(clock),
+  private fun admitOffenderOnNewBooking(confirmation: Confirmation, prisonNumber: String): ConfirmArrivalResponse {
+    val response = prisonService.admitOffenderOnNewBooking(prisonNumber, confirmation.detail)
+
+    whenExpectedArrival(confirmation) { arrivalId, detail ->
```

Comment:
we only save expected arrivals as we don't need to remove unexpected arrivals from the list of incoming arrivals for that day